src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,265 @@
+            } catch (DukeException e) {
+                System.out.println(DIVLINE + "Target task does not exist or you have specified nothing.");
+                System.out.print(DIVLINE);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            break;
+        default:
+            throw new DukeException("Illegal keyword.");
+        }
+         return;
+    }
+
+    /**
+     * */
+    private static void displayList() throws DukeException {
+        if (taskList.isEmpty()) {
+            throw new DukeException("Empty list. Nothing to be displayed.");
+        }
+        System.out.print(DIVLINE);
+        System.out.println("\tHere are the tasks in your list:");
+        // Create an iterator for the list
+        Iterator<Task> iter = taskList.iterator();
+        // Displaying the values after iterating through the list
+        int counter = 0;
+        while (iter.hasNext()) {
+            counter++;
+            System.out.println("\t" + counter + "." + iter.next().toString());
+        }
+        System.out.print(DIVLINE);
+    }
+
+    private static void addDeadlineToList(String description) throws DukeException, IOException {
+        if (description.startsWith(" ")){
+            String ddlDscp = description.substring(1);
+            int splitPoint = ddlDscp.indexOf("/by");
+            if (splitPoint==-1){
+                throw new DukeException("Illegal Deadline prompt detected.");
+            }
+            Deadline newDdl = new Deadline(ddlDscp.substring(0, splitPoint - 1),
+                    ddlDscp.substring(splitPoint + 4));
+            taskList.add(newDdl);
+            writeToFile(initializeTextToWrite());
+            printAddSuccessMessage(newDdl);
+        } else {
+            throw new DukeException("Illegal Deadline prompt detected.");
+        }
+    }
+
+    private static void addEventToList(String description) throws DukeException, IOException {
+        if (description.startsWith(" ")){
+            String evtDscp = description.substring(1);
+            int splitPoint = evtDscp.indexOf("/at");
+            if (splitPoint==-1){
+                throw new DukeException("Illegal Event prompt detected.");
+            }
+            Event newEvt = new Event(evtDscp.substring(0, splitPoint - 1),
+                    evtDscp.substring(splitPoint + 4));
+            taskList.add(newEvt);
+            writeToFile(initializeTextToWrite());
+            printAddSuccessMessage(newEvt);
+        } else {
+            throw new DukeException("Illegal Event prompt detected.");
+        }
+    }
+
+    private static void addTodoToList(String description) throws DukeException, IOException {
+        if (description.startsWith(" ")) {
+            Task newTodo = new Todo(description.substring(1));
+            taskList.add(newTodo);
+            writeToFile(initializeTextToWrite());
+            printAddSuccessMessage(newTodo);
```

Comment:
Here, there are two methods being used, except that steps to task creation and addition to task list are explicitly written. You could consider incorporating [SLAP](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard).